onset_maproom/calc.py

```diff
@@ -13,6 +13,45 @@ def read_zarr_data(zarr_path):
 # Growing season functions
 
 
+def estimate_sm(
```

Comment:
Rename `estimate_soil_moisture`, or just `soil_moisture`?

If you expect this function will be useful beyond the cessation date calculation, I think it should have a more self-explanatory name. If it's just an internal detail of the cessation date implementation then I don't care too much.

Reply:
This is actually the basic algo of the Ingrid water balance. See W(t)= [section](http://iridl.ldeo.columbia.edu/dochelp/Documentation/details/index.html?func=:Water_Balance). The Ingrid water balance just does a lot more stuff to what goes into this basic algo. So it actually makes sense to call this water_balance that could be called by other more elaborated water_balance functions later. Or a same function with options that trigger this or that element of the water_balance?

You may see that the recurrence relationship involved ET (or ETcrop_red) that is actually dependent on SM(t-1) so it means intervention in the loop. Except that actually in Ingrid, I loop only to get SM and ETcrop_red is actually retrieved vectorially after the fact (as it becomes an output of water_balance among several others besides SM). So the loop might be fine as it is.

I guess my point is that it would make sense to name the function water_balance, like Ingrid's, and that it can expand later with the other parts of the algo, that can be triggered with optional arguments (I seem to see that this is a fine thing to do in Xarray & Co); with the slight drawback that some rewriting might be needed to accommodate those options.

Makes sense?

And also in that case that function will likely be used by much more than cessation_date. It will probably be desired by PIs sooner than later...

Reply:
sounds good.

Reply:
sorry... moving in small steps here. Should I make the output a dataset then of which soil_moisture which be a variable? Assuming that later, there will be more variables output in the dataset created by water_balance function?

Reply:
Sure, if you anticipate that it will return multiple variables in the future, it would be good to return a Dataset so that won't have to be a breaking change.

Out of curiosity, what other variables?

Reply:
See the list in the ingrid func doc, just before the Notes. Below links to examples. Basically outputs all the different elements of the water balance. You can also look at what Maprooms using this look like (it's slow).

http://iridl.ldeo.columbia.edu/maproom/Water/BAWKP/Agriculture/Bicol_WB_Dry.html?region=irids%3Ahome%3Aremic%3APhilippines%3ABicol%3AsoilTypes%3Agid%40185%3Ads

http://maproom.meteorwanda.gov.rw/maproom/Agriculture/Histo_Agri/WB_BB_MAM.html?region=bb%3A29.56875%3A-2.6812500000000004%3A29.606250000000003%3A-2.6437500000000003%3Abb